plasma-pass: init at 1.2.0

jb55 opened the pull request:
Signed-off-by: Matthias Beyer <mail@beyermatthias.de>
Reviewed-by: William Casarin <jb55@jb55.com>
Link: https://lists.sr.ht/~andir/nixpkgs-dev/%3C20210426121952.17969-1-mail@beyermatthias.de%3E

legendofmiracles commented on pkgs/tools/security/plasma-pass/default.nix:
```diff
@@ -0,0 +1,39 @@
+  nativeBuildInputs = [ cmake extra-cmake-modules ];
+
+  meta = with lib; {
+    description = "a Plasma applet to access password from pass, the standard UNIX password manager";
```
```suggestion
    description = "A Plasma applet to access passwords from pass, the standard UNIX password manager";
```

jb55 commented:
***@***.*** commented on this pull request.
>```suggestion
>    description = "A Plasma applet to access passwords from pass, the standard UNIX password manager";
>```

thanks, updated

SuperSandro2000 commented on pkgs/tools/security/plasma-pass/default.nix:
```diff
@@ -0,0 +1,39 @@
+  pname = "plasma-pass";
+  version = "1.2.0";
+
+  src = fetchgit {
```
Please use fetchFromGitLab.

SuperSandro2000 commented on pkgs/tools/security/plasma-pass/default.nix:
```diff
@@ -0,0 +1,39 @@
+
+  meta = with lib; {
+    description = "A Plasma applet to access passwords from pass, the standard UNIX password manager";
+    homepage = "https://github.com/KDE/plasma-pass";
```
```suggestion
    homepage = "https://invent.kde.org/plasma/plasma-pass";
```

SuperSandro2000 commented on pkgs/tools/security/plasma-pass/default.nix:
```diff
@@ -0,0 +1,39 @@
+    description = "A Plasma applet to access passwords from pass, the standard UNIX password manager";
+    homepage = "https://github.com/KDE/plasma-pass";
+    license = licenses.lgpl21Plus;
+    maintainers = with maintainers; [ matthiasbeyer ];
```
Please do not add other people.

matthiasbeyer replied:
Nobody adds other people here.

SuperSandro2000 replied:
That's wrong. Someone can't just forward someones patch to nixpkgs from some other repo without consent of the author/maintainer. If you are opening this with your 2nd GitHub account please make this clear somewhere.

If this behavior would be allowed then packages could be associated to maintainers which never agreed with that and create a false hint to someone who is interested in the package.

You clearly have access to a GitHub account, so please use it to add packages you want to have in nixpkgs instead of proxying over 2 conners.

matthiasbeyer replied:
> Someone can't just forward someones patch to nixpkgs from some other repo without consent of the author/maintainer

Nobody did. There's the `Signed-off-by` line in the patch. Case closed.

> You clearly have access to a GitHub account, so please use it to add packages/maintainerships you want to have in nixpkgs instead of proxying over 2 conners.

I submit patches the way I like, not the way you would like to command me to.

lschuermann replied:
> That's wrong. Someone can't just forward someones patch to nixpkgs from some other repo without consent of the author/maintainer.

I think you misspelled "free software" there. I suggest reading up on the [MIT license](https://github.com/NixOS/nixpkgs/blob/master/COPYING).

SuperSandro2000 replied:
> Nobody did. There's the Signed-off-by line in the patch. Case closed.

Anyone can add this. I can also submit a patch in your name with that line.

> I think you misspelled "free software" there.

Well, of course I could do that but we do not accept users adding other people as maintainers for obvious reasons.

JohnAZoidberg replied:
If @matthiasbeyer says he agrees, then it's ok. No matter how the patch was submitted.

matthiasbeyer committed:
plasma-pass: init at 1.2.0

Reviewed-by: William Casarin <jb55@jb55.com>
Signed-off-by: Matthias Beyer <mail@beyermatthias.de>
Link: https://lists.sr.ht/~andir/nixpkgs-dev/%3C20210427123311.517-1-mail@beyermatthias.de%3E

jb55 commented:
On Tue, Apr 27, 2021 at 05:30:29AM -0700, Sandro wrote:
>@SuperSandro2000 commented on this pull request.
>+  meta = with lib; {
>+    description = "A Plasma applet to access passwords from pass, the standard UNIX password manager";
>+    homepage = "https://github.com/KDE/plasma-pass";
>+    license = licenses.lgpl21Plus;
>+    maintainers = with maintainers; [ matthiasbeyer ];
>
>That's wrong. Someone can't just forward someones patch to nixpkgs from
>some other repo without consent of the author/maintainer. If you are
>opening this with your 2nd GitHub account please make this clear
>somewhere.

There are many people who prefer not to use GitHub for whatever reason,
including myself. We are forced to due to nixpkgs workflows. We have set
up a mailing list[1] to submit and review patches before submitting them
here. I do not see the problem with an extra layer of review before
submission here.

>If this behavior would be allowed then packages could be associated to
>maintainers which never agreed with that and create a false hint to
>someone who is interested in the package.

I really don't see a problem here. The person who is interested in the
package is typically the person in the maintainer field, I don't mind
shepharding the PR in the meantime for people who rather not use GitHub.

Cheers,
Will

SuperSandro2000 commented:
> I do not see the problem with an extra layer of review before
submission here.

That's not the problem.

> I really don't see a problem here.

I could submit a package with you in the maintainer field where you don't want to be and I think we shouldn't allow that.

jb55 commented:
On Tue, Apr 27, 2021 at 01:01:36PM -0700, Sandro wrote:
>> I do not see the problem with an extra layer of review before
>submission here.
>
>That's not the problem.
>
>> I really don't see a problem here.
>
>I could submit a package with you in the maintainer field where you don't want to be and I think we shouldn't allow that.

We already have the discourse inbox forum for this exact purpose. People
have been submitting patches on behalf of other people for awhile. git
explicitly supports this use case which is why there is separate
committer and author fields. 

Are you stating that I am untrustworthy? I have been part of this
community since 2015, who are you?

Ma27 reviewed (approved):
* We're talking about a maintainer entry - something that's trivially changeable, not - idk - the paying side of a financial transaction.
* The email, @matthiasbeyer used on sourcehut to file the patch is the same as in the maintainer entry they specified. The majority of spoofing attempts for emails can be detected via DKIM. Of course there's a remaining risk, but GitHub accounts aren't "unhackable" either.
* This patch goes through the PR queue just like any other patch does.
* Even though we use GitHub as primary platform, it's a good thing to provide reasonable alternatives for contributors. In some cases that's the only choice btw, for instance GitHub [blocked folks from e.g. Iran in the past](https://twitter.com/natfriedman/status/1155311122137804801) (I'm not a lawyer, but the fact that this has happened - for private repos only - means it can happen again for affected users, also for public repos. In other words, I wouldn't trust GitHub and think that alternatives are a good thing because of that).
* Thanks to @jb55 who acts as a proxy here!
* Last but not least, the diff looks fine as well :)

jb55 commented:
On Thu, Apr 29, 2021 at 02:04:06PM -0700, Maximilian Bosch wrote:
>@Ma27 approved this pull request.
>
>* We're talking about a maintainer entry - something that's trivially changeable, not - idk - the paying side of a financial transaction.
>* The email, @matthiasbeyer used on sourcehut to file the patch is the same as in the maintainer entry they specified. The majority of spoofing attempts for emails can be detected via DKIM. Of course there's a remaining risk, but GitHub accounts aren't "unhackable" either.
>* This patch goes through the PR queue just like any other patch does.
>* Even though we use GitHub as primary platform, it's a good thing to provide reasonable alternatives for contributors. In some cases that's the only choice btw, for instance GitHub [blocked folks from e.g. Iran in the past](https://twitter.com/natfriedman/status/1155311122137804801).
>* Thanks to @jb55 who acts as a proxy here!
>* Last but not least, the diff looks fine as well :)

Thanks! Glad to see more support for this workflow. Many people have
contacted me and are happy they can contribute without needing GitHub,
hopefully we can keep it going without too much friction.
